test/Feature/HLSLLib/fmod.16.test

```diff
@@ -0,0 +1,77 @@
+#--- source.hlsl
+StructuredBuffer<half4> In0 : register(t0);
+StructuredBuffer<half4> In1 : register(t1);
+
+RWStructuredBuffer<half4> Out0 : register(u2);
+
+
+[numthreads(1,1,1)]
+void main() {
+
+Out0[0] = fmod(In0[0], In1[0]);
+Out0[1] = half4(fmod(In0[1].xyz, In1[1].xyz), fmod(In0[1].w, In1[1].w));
+Out0[2] = half4(fmod(In0[2].xy, In1[2].xy), fmod(In0[2].zw, In1[2].zw));
+Out0[3] = fmod(half4(10.5, -99.5, 5, 0.25), half4(1, -3, 0.25, 5));
+  }
+//--- pipeline.yaml
+
+---
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [1, 1, 1]
+Buffers:
+  - Name: In0
+    Format: Float16
+    Stride: 8
+    Data: [0x4940, 0xd638, 0x4500, 0x3400, 0, 0x6056, 0x4940, 0xd638, 0x4500, 0x3400, 0, 0x6056]
+    # 10.5, -99.5, 5, 0.25, 0, 555, 10.5, -99.5, 5, 0.25, 0, 555 
+  - Name: In1
+    Format: Float16
+    Stride: 8
+    Data: [0x3c00, 0xc200, 0x3400, 0x4500, 0x4900, 0x4100, 0x3c00, 0xc200, 0x3400, 0x4500, 0x4900, 0x4100]
+    # 1.1, -3, 0.25, 5, 10, 2.5, 1, -3, 0.25, 5, 10, 2.5
+  - Name: Out0
+    Format: Float16
+    Stride: 8
+    ZeroInitSize: 32
+  - Name: ExpectedOut0
+    Format: Float16
+    Stride: 8
+    Data: [ 0x3800, 0xB780, 0x0, 0x3400, 0x0, 0x0, 0x3800, 0xB780, 0x0, 0x3400, 0x0, 0x0, 0x3800, 0xB800, 0x0, 0x3400 ]
+    # 0.5, -0.46875, 0, 0.25, 0, 0, 0.5 -0.46875, 0, 0.25, 0, 0, 0.5, -0.46875, 0, 0.25
```

Comment:
I expected 10.5 mod 1.1 to be 0.6, I'm surprised 0.5 is the answer? It also doesn't seem like epsilon is high enough to tolerate this error, I'm surprised this is passing.
Also shocked at how far off -99.5 mod -3 is from .5, should we add `-Gis` to the compilation options perhaps to improve precision?

Reply:
Yeah the comment on the value was wrong, it was 1 not 1.1; Added Gis and have set the reuslt to -.5